network-api/networkapi/wagtailpages/pagemodels/research_hub/forms.py

```diff
@@ -0,0 +1,65 @@
+from django import forms
+from django.utils.translation import pgettext_lazy
+
+from networkapi.wagtailpages import utils
+from networkapi.wagtailpages.pagemodels import profiles as profile_models
+from networkapi.wagtailpages.pagemodels.research_hub import detail_page, taxonomies
+
+
+def _get_author_options():
+    author_profiles = profile_models.Profile.objects.filter_research_authors()
+    author_profiles = utils.localize_queryset(author_profiles)
+    return [(author_profile.id, author_profile.name) for author_profile in author_profiles]
+
+
+def _get_topic_options():
+    topics = taxonomies.ResearchTopic.objects.all()
+    topics = utils.localize_queryset(topics)
+    return [(topic.id, topic.name) for topic in topics]
+
+
+def _get_region_options():
+    regions = taxonomies.ResearchRegion.objects.all()
+    regions = utils.localize_queryset(regions)
+    return [(region.id, region.name) for region in regions]
+
+
+def _get_year_options():
+    dates = detail_page.ResearchDetailPage.objects.dates(
+        "original_publication_date",
+        "year",
+        order="DESC",
+    )
+    year_options = [(date.year, date.year) for date in dates]
+    empty_option = (
+        "",
+        pgettext_lazy("Option in a list of years", "Any"),
+    )
+    return [empty_option] + year_options
```

Comment:
Moved all of these to this module (they don't depend on the Library Page) and changed the output to tuples (the expected format on the `choices` parameter

Reply:
Thanks Jhonatan! This makes sense to me, 👍